File: JbmrksIOs/JbmrksIOs/Services/TokenStorage.swift
//
//  TokenStorage.swift
//  JbmrksIOs
//
//  Native Swift TokenStorage protocol (temporary)
//

import Foundation

protocol TokenStorage {
    nonisolated func saveAccessToken(token: String)
    nonisolated func getAccessToken() -> String?
    
    nonisolated func saveRefreshToken(token: String)
    nonisolated func getRefreshToken() -> String?
    
    nonisolated func savePortalUrl(url: String)
    nonisolated func getPortalUrl() -> String?
    
    nonisolated func saveTokenExpiry(expiresIn: Int)
    nonisolated func getTokenExpiry() -> Int64?
    
    nonisolated func isTokenExpired() -> Bool
    
    nonisolated func clear()
}

File: shared/src/commonMain/kotlin/storage/TokenStorage.kt
package com.example.jbmarks.shared.storage

/**
 * Interface for token storage operations
 * Handles OAuth tokens and portal URL storage
 * 
 * NOTE: This interface uses regular (non-suspend) functions to allow Swift implementations.
 * For async operations, implementations should handle asynchrony internally.
 */
interface TokenStorage {
    fun saveAccessToken(token: String)
    fun getAccessToken(): String?
    
    fun saveRefreshToken(token: String)
    fun getRefreshToken(): String?
    
    fun savePortalUrl(url: String)
    fun getPortalUrl(): String?
    
    fun saveTokenExpiry(expiresIn: Int)
    fun getTokenExpiry(): Long?
    
    fun isTokenExpired(): Boolean
    
    fun clear()
}
